sbhs/views.py

```diff
@@ -318,15 +318,31 @@ def slot_new(request):
             form = SlotCreationForm(request.POST)
             if form.is_valid():
                 new_slot = form.save(commit=False)
-
                 new_slot_date = new_slot.start_time.date()
                 new_slot_time = new_slot.start_time.time()
+                prev_hour = dt.combine(
+                            new_slot_date,time(
+                                new_slot_time.hour,00)) - timedelta(hours=1)
+                prev_slot = slot_history.filter(start_time__date=new_slot_date,
+                                                start_time__time=prev_hour)
+                next_hour = dt.combine(
+                                new_slot_date,time(
+                                new_slot_time.hour,00)) + timedelta(hours=1)
+                next_slot = slot_history.filter(start_time__date=new_slot_date,
```

Comment:
What is difference between prev_hour and next_hour, also next_plot and prev_slot?

Reply:
**prev_hour** is _**"previous hour"**_ of **current hour** and  **next_hour** is _**"next_hour"**_ of **current_hour**. **prev_slot** is the query to find any booked slot on **prev_hour** and **next_slot** is to find the booked slot for **next_hour**. Basically There should not be any **consecutive booking** of slots. There should be at least 1 hour gap between slot booking.

Reply:
So have you tested booking after an hour?
Does it work?
B'coz I think as per the code the gap is atleast of 2 hours.
Kindly check.

Reply:
Yeah it works.

Reply:
I am still not convince with the code, there are many things like,
1. form is saved first and then date is validated
2. validation can be in forms or a separate function, as the view as become very huge and complex
3. date and time in filter is matched by equal to and no less than or greater to...
Would like to see the working before finalizing this.